src/AI/Muriel.java

```diff
@@ -955,20 +959,26 @@ private void queueUnitProductionActions(GameMap gameMap)
     } // ~while( still choosing units to build )
 
     // Build infantry from any remaining facilities.
-    UnitModel infModel = myCo.getUnitModel(UnitModel.TROOP);
-    final int infCost = myCo.getCost(infModel);
-    while( (budget >= infCost) && (CPI.availableUnitModels.contains(infModel)) )
+    UnitModel infModel = myCo.cos[0].getUnitModel(UnitModel.TROOP);
```

Comment:
I do like `Army.getShoppingList(XYCoord)`, since that gives you the knowledge of what CO to look at.
The awkward bit here is that (per past-me's "modularity" concerns) `GameInstance` isn't very visible (and injecting that dependency could be a little awkward).
It's probably a good idea to feed AIs the rules on construction, and that does seem like a reasonable "master location" for the list of models.

Reply:
Taking a gander, it looks like there's nowhere we invoke `Army.cos[0].getShoppingList()`, so I don't think adding it to Army's interface makes sense right now?

Reply:
Fair enough about `getShoppingList`. Still seems... awkward to have to query army for a CO to get a unit model that is defined independently of any CO or army though.